src/AutoRest.CSharp/Common/Generation/Types/CSharpType.cs

```diff
@@ -163,12 +214,16 @@ internal static CSharpType FromSystemType(Type type, string defaultNamespace, So
         internal static CSharpType FromSystemType(BuildContext context, Type type)
             => FromSystemType(type, context.DefaultNamespace, context.SourceInputModel);
 
-        public CSharpType GetNonNullable()
+        public bool IsCollectionType()
         {
-            if (!IsNullable)
-                return this;
+            if (!IsFrameworkType)
+                return false;
 
-            return IsFrameworkType ? new CSharpType(FrameworkType, false, Arguments) : new CSharpType(Implementation, Namespace, Name, IsValueType, IsEnum, false, Arguments);
+            return FrameworkType.Equals(typeof(IList<>)) ||
+                FrameworkType.Equals(typeof(IEnumerable<>)) ||
+                FrameworkType == typeof(IReadOnlyList<>) ||
+                FrameworkType.Equals(typeof(IDictionary<,>)) ||
+                FrameworkType == typeof(IReadOnlyDictionary<,>);
         }
```

Comment:
since we have another `IsCollectionType(CSharpType)` method in `TypeFactory` as well and it is widely used in a lot of places, do we really need this new one?

Reply:
No, and it is not used anywhere. Will remove.

Reply:
Removed